src/main/java/core/basesyntax/BlockingQueue.java

```diff
@@ -12,16 +12,31 @@ public BlockingQueue(int capacity) {
     }
 
     public synchronized void put(T element) throws InterruptedException {
-        // write your code here
+        while (queue.size() == capacity) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                throw new RuntimeException("Something went wrong!", e);
+            }
```

Comment:
you can remove try catch, you already have `throws InterruptedException` in the signature of method